src/ssl.c

```diff
@@ -25513,6 +25568,11 @@ int wolfSSL_PEM_write_bio_RSAPrivateKey(WOLFSSL_BIO* bio, WOLFSSL_RSA* key,
 
 
     pkey = wolfSSL_PKEY_new_ex(bio->heap);
+    if (pkey == NULL) {
+        WOLFSSL_MSG("wolfSSL_PKEY_new_ex failed");
+        return SSL_FAILURE;
+    }
+
```

Comment:
It looks like with `WOLFSSL_KEY_GEN` defined there are cases where `pkey` isn't free'd below.